src/file.cpp

```diff
@@ -206,7 +207,7 @@ File *File::unserialize(string *d)
     delete fp;
 
     file->name.assign(name, namelen);
-    file->localname.editStringDirect()->assign(localname, localnamelen);
+    file->localname = LocalPath::fromPlatformEncoded(std::string(localname, localnamelen));
```

Comment:
this could use a new `fromPlatformEncoded(const char *, unsigned short)` constructor

Reply:
I think it's best not to have the constructor overloaded for char* and wchar_t* (or string and wstring) because that allows accidental use of the wrong type of string, without the double check of the 'platform encoded' reminder.